src/main/java/org/cyclops/iconexporter/client/gui/ScreenIconExporter.java

```diff
@@ -89,35 +107,79 @@ public String serializeNbtTag(INBT tag) {
 
         // Add fluids
         for (Map.Entry<RegistryKey<Fluid>, Fluid> fluidEntry : ForgeRegistries.FLUIDS.getEntries()) {
-            tasks.set(tasks.get() + 1);
-            String subKey = "fluid:" + fluidEntry.getKey().getLocation();
-            exportTasks.add((matrixStack) -> {
-                taskProcessed.set(taskProcessed.get() + 1);
-                signalStatus(tasks, taskProcessed);
-                fill(matrixStack, 0, 0, scaleModifiedRounded, scaleModifiedRounded, BACKGROUND_COLOR);
-                ItemRenderUtil.renderFluid(this, matrixStack, fluidEntry.getValue(), scaleModified);
-                ImageExportUtil.exportImageFromScreenshot(baseDir, subKey, this.width, this.height, this.scaleImage, BACKGROUND_COLOR_SHIFTED);
-            });
+            ResourceLocation location = fluidEntry.getKey().getLocation();
+            if (filter.test(location)) {
+                tasks.set(tasks.get() + 1);
+                String subKey = "fluid:" + location;
+                //Test if fluid is opaque
+                if (RenderTypeLookup.canRenderInLayer(fluidEntry.getValue().getDefaultState(), RenderType.getSolid())) {
+                    exportTasks.add((matrixStack, bImage) -> {
```

Comment:
There's quite a bit of code duplication in these tasks. Could this be abstracted? Otherwise this significantly complicates maintenance effort in future updates.